connectors/src/resources/connector_res.ts

```diff
@@ -0,0 +1,28 @@
+import type { ConnectorProvider } from "@dust-tt/types";
+import type { ModelStatic } from "sequelize";
+
+import { BaseResource } from "@connectors/resources/base_res";
+import { ConnectorModel } from "@connectors/resources/storage/models/connector_model";
+import type { ReadonlyAttributesType } from "@connectors/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ConnectorResource
+  extends ReadonlyAttributesType<ConnectorModel> {}
+export class ConnectorResource extends BaseResource<ConnectorModel> {
+  static model: ModelStatic<ConnectorModel> = ConnectorModel;
+
+  static async listByType(type: ConnectorProvider) {
```

Comment:
TBH I am not a big fan of this example of "listByType".
Then you want to add pagination to this method, and another criteria that becomes an optional one, and over time this function has 10 arguments and becomes really hard to read.

Can't we have a `fetchAll()` function like Sequelize offers?

Reply:
Every time I dealt with such helper function, life became a nightmare over time, while explicit filtering was a blessing.

Reply:
The goal of the resource approach is to never expose SQL clause structure outside of it. I implemented this function since as of now, Connectors is pretty simple and does not need anything else than filtering on types. If we need to have 10 parameters it probably means that we need different methods.